wdl-analysis/src/types/v1.rs

```diff
@@ -692,6 +693,33 @@ impl<'a, C: EvaluationContext> ExprTypeEvaluator<'a, C> {
         self.placeholders -= 1;
     }
 
+    fn invalid_placeholder_option(
+        ty: &Type,
+        _span: Span,
+        option: Option<PlaceholderOption>,
+    ) -> Diagnostic {
+        let message = match option {
+            Some(PlaceholderOption::Sep(_)) => format!(
+                "type mismatch for placeholder option `sep`: expected type `Array[P]` where P: any primitive type or Union, but found `{}`",
+                ty
+            ),
+            Some(PlaceholderOption::Default(_)) => format!(
+                "type mismatch for placeholder option `default`: expected an optional primitive type, but found `{}`",
+                ty
+            ),
+            Some(PlaceholderOption::TrueFalse(_)) => format!(
+                "type mismatch for placeholder option `truefalse`: expected type `Boolean`, but found `{}`",
+                ty
+            ),
+            None => format!("cannot coerce type `{}` to `String`", ty),
+        };
+        Diagnostic::error(message) // No span here
```

Comment:
We should definitely be attaching a span to the error, otherwise it will default to the end of the file and that's not what we want.

```suggestion
        Diagnostic::error(message).with_label(format!("this is type `{ty}`"), span)
```